code/modules/antagonists/heretic/items/heretic_grenade.dm

```diff
@@ -0,0 +1,111 @@
+/*!
+ * Contains Heretic grenades
+ * They spread rust and obliterate borgs/mechs
+ */
+
+/obj/item/grenade/chem_grenade/rust_sower
+	name = "\improper Rust sower"
+	desc = "A nifty little thing that explodes into rust. Causes borgs and mechs to get utterly obliterated"
+	possible_fuse_time = list("5")
+	stage = GRENADE_READY
+	base_icon_state = "rustgrenade"
+	grenade_arm_sound = 'sound/items/weapons/rust_sower_armbomb.ogg'
+	grenade_sound_vary = FALSE
+
+/obj/item/grenade/chem_grenade/rust_sower/update_icon_state()
+	. = ..()
+	if(active)
+		icon_state = "[base_icon_state]_active"
+	else
+		icon_state = base_icon_state
+
+/obj/item/grenade/chem_grenade/rust_sower/Initialize(mapload)
+	. = ..()
+	var/obj/item/reagent_containers/cup/beaker/large/beaker_one = new(src)
+	var/obj/item/reagent_containers/cup/beaker/large/beaker_two = new(src)
+
+	beaker_one.reagents.add_reagent(/datum/reagent/heretic_rust, 50)
+	beaker_one.reagents.add_reagent(/datum/reagent/potassium, 50)
```

Comment:
Would proobably make more sense for this to have and heat smoke powder rather than the ingredients for a smoke reaction? This feels... silly.